File: PB - More Exercises Python/2.0 Conditional Statements - More Exercises/09. Fuel Tank - Part 2.py
type_fuel, quantity, club_card = input(), float(input()), input()
fuel_price_dict = {'Gasoline' : 2.22,'Diesel' : 2.33,'Gas' : 0.93}
discount = {'Gasoline' : 0.18,'Diesel' : 0.12,'Gas' : 0.08}
subtotal_price, total_price = 0

if club_card == 'Yes':
    subtotal_price = fuel_price_dict[type_fuel] - discount[type_fuel]
    total_price = subtotal_price * quantity
else:
    total_price = fuel_price_dict[type_fuel] * quantity

if 20 <= quantity <= 25:
    total_price *= 0.92
elif quantity > 25:
     total_price *= 0.90

print(f'{total_price:.2f} lv.')



'''
Gas
30
Yes
'''

File: PB - More Exercises GO/2.0 Conditional Statements - More Exercises/09. Fuel Tank - Part 2.go
package main

import "fmt"

func main() {
	var quantity float64
	var typeFuel, clubCard string
	fmt.Scan(&typeFuel, &quantity, &clubCard)

	fuelPriceDict := map[string]float64{"Gasoline": 2.22, "Diesel": 2.33, "Gas": 0.93}
	discount := map[string]float64{"Gasoline": 0.18, "Diesel": 0.12, "Gas": 0.08}
	subtotalPrice := 0.0
	totalPrice := 0.0

	if clubCard == "Yes" {
		subtotalPrice = fuelPriceDict[typeFuel] - discount[typeFuel]
		totalPrice = subtotalPrice * quantity
	} else {
		totalPrice = fuelPriceDict[typeFuel] * quantity
	}

	if 20 <= quantity && quantity <= 25 {
		totalPrice *= 0.92
	} else if quantity > 25 {
		totalPrice *= 0.90
	}

	fmt.Printf("%.2f lv.", totalPrice)

}

/*
name   :09. Fuel Tank - Part 2
input  :Gas 30 Yes
output :22.95 lv.
*/
